gluster-exporter/main.go

```diff
@@ -41,8 +46,51 @@ func registerMetric(name string, fn func(), intervalSeconds int64) {
 	glusterMetrics = append(glusterMetrics, glusterMetric{name: name, fn: fn, intervalSeconds: time.Duration(intervalSeconds)})
 }
 
+func getGlusterdWorkdir() string {
+        if( *glusterdWorkdir != "" && *glusterMgmt == "glusterd1" ) {
+		filepath = *glusterdWorkdir + "/glusterd.info"
+		keywordID = "UUID"
+        } else if( *glusterdWorkdir != "" && *glusterMgmt == "glusterd2") {
+                filepath = *glusterdWorkdir + "/uuid.toml"
+                keywordID = "peer-id"
+        } else if( *glusterMgmt == "glusterd2") {
+                filepath = defaultGd2Workdir + "/uuid.toml"
+                keywordID = "peer-id"
+        } else {
+		filepath = defaultGd1Workdir + "/glusterd.info"
+		keywordID = "UUID"
+        }
+        return filepath
+}
+
 func getPeerID() string {
-	return *peerid
+        if(peerID == "") {
+		filepath = getGlusterdWorkdir()
```

Comment:
`getGlusterdWorkdir` should only return workdir.

    workdir = getGlusterdWorkingdir()
    keywordID := "UUID"
    filepath := workdir + "/glusterd.info"
    if *glusterMgmt == "glusterd2" {
        keywordID = "peer-id"
        filepath = workdir + "/uuid.toml"
    }